atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -344,6 +344,40 @@ void WebFrame::SetIsolatedWorldHumanReadableName(int world_id,
       world_id, blink::WebString::FromUTF8(name));
 }
 
+void WebFrame::SetIsolatedWorldInfo(int world_id, mate::Arguments* args) {
+  if (args->Length() < 1) {
+    args->ThrowError("Invalid args");
+    return;
+  }
+
+  v8::Local<v8::Value> val;
+  args->GetNext(&val);
+
+  mate::Dictionary options;
+  if (!mate::ConvertFromV8(args->isolate(), val, &options)) {
+    args->ThrowError("Must pass valid object");
+    return;
+  }
+
+  std::string origin, csp, name;
+  options.Get("securityOrigin", &origin);
+  options.Get("csp", &csp);
+  options.Get("name", &name);
+
+  if (!csp.empty() && origin.empty()) {
+    args->ThrowError(
+        "If csp is spicified, securityOrigin should also be specified");
+    return;
+  }
+
+  if (!origin.empty())
```

Comment:
Not necessary to check the emptiness of the types for the following here, the blink apis will handle it.